src/main/kotlin/no/nav/syfo/cronjob/VerifyBehandlereForKontorCronjob.kt

```diff
@@ -297,9 +297,31 @@ class VerifyBehandlereForKontorCronjob(
             } else if (existingBehandlereWithSameHprId.size == 1) {
                 val existingBehandler = existingBehandlereWithSameHprId[0]
                 val hprBehandlerFnr = syfohelsenettproxyClient.finnBehandlerFraHpr(behandlerFraAdresseregisteretHprId)?.fnr
-                if (hprBehandlerFnr == null || hprBehandlerFnr != existingBehandler.personident) {
-                    log.warn("VerifyBehandlereForKontorCronjob: Mismatched personident: ${existingBehandler.behandlerRef}")
+                val existingBehandlerFnr = existingBehandler.personident
+                val doUpdate = if (hprBehandlerFnr == null) {
+                    false
+                } else if (existingBehandlerFnr == null) {
+                    behandlerService.updateBehandlerPersonident(
+                        behandlerRef = existingBehandler.behandlerRef,
+                        personident = hprBehandlerFnr,
+                    )
```

Comment:
Er det ikke litt dumt om vi gjør en oppdatering i databasen i en kodeblokk som egentlig skal sette en variabel til true eller false? Kunne vi løst det på en annen måte?

Reply:
Ja, går an å regne ut to booleans: en for om personident'en skal oppdateres, og en for om resten skal oppdateres. Blir kanskje bedre - kan prøve.

Reply:
Update: Tror jeg klarte å gjøre det litt mindre innfløkt 👍